### PR TITLE
AST-2021 - Row block keeps changing to Group block in the Block Editor

### PR DESCRIPTION
**Pull request by GauravKhupse**
### Description

The row block in the block editor keeps changing to the group block and causing the layout issue in the editor itself.

This issue does not occur on the frontend but editor only.

Steps To Reproduce: 
Create a page and add the row block
Add some images to the row block and publish the page
Check the front; it looks good.
Now come back to the editor and refresh the page, the row block is changed to the group block, and the layout is different.
You will notice horizontal images as vertical.
Try changing to the default theme, and the issue is gone.

### Screenshots

https://d.pr/v/iwZBOT

### Checklist:
- [x] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: inc/assets/js/column-block-compatibility.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ wp.hooks.addFilter(
 			return attributes;
 		}
 
-		if (blockType.name == 'core/group' && undefined != attributes.layout && false == attributes.layout.inherit ) {
+		if (blockType.name == 'core/group' && undefined != attributes.layout && 'flex' === attributes.layout.type) {
```
Comment (imnavanath) [inline inc/assets/js/column-block-compatibility.js:42]: Add block type condition top, and beneath keep the condition `false == attributes.layout.inherit` as its required for Group block.

Comment (GauravKhupse) [reply inc/assets/js/column-block-compatibility.js:42]: Changes done and also checked with all row types with flex shows.

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,6 @@
+v3.9.0
```
Comment (imnavanath) [inline changelog.txt:1]:
```suggestion
v3.9.1
```

**Review by imnavanath — APPROVED** [review]

**Review by yssambare12 — APPROVED** [review]

File: changelog.txt (hunk 2 above)
Comment (GauravKhupse) [reply changelog.txt]: changes done!

**Pull request merged**